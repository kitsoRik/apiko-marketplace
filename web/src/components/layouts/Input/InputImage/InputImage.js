import React, { useState } from 'react';

import "./InputImage.scss";

const InputImage = ({ className, setFile, ...props }) => {

    const [isDrag, setIsDrag] = useState(false);
    const [image, setImage] = useState(null);

    const onChange = (e) => {
        const file = e.target.files[0];

        if (!file) return;

        let reader = new FileReader();

        reader.onload = (event) => {
            setImage(event.target.result);
        }

        setFile(file);
        reader.readAsDataURL(file);
    }

    const onClick = (e) => {
        //ref.current.click();
    }

    const ref = React.createRef();

    return (
        <div
            className={`input-image ${className ?? ""}`}
            onClick={onClick}
            isDrag={isDrag ? "true" : null}
            {...props}
        >
            <input
                ref={ref}
                type="file"
                onChange={onChange}
                onDragEnter={() => setIsDrag(true)}
                onDragLeave={() => setIsDrag(false)}
            />
            <div className="input-image-container">
                <img src={image} alt="Unknown" />
            </div>
        </div>
    )
};

export default InputImage;