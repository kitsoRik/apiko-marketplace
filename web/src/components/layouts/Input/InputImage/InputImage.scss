.input-image {
	height: 92px;
	width: 92px;

	position: relative;

	background-color: #e4e4e4;

	border-radius: 4px;

	cursor: pointer;

	&:hover {
		background-color: #d3d3d3;
	}

	&[has-file="false"] {
		&::before,
		&::after {
			content: "";

			height: 4px;
			width: 38px;

			position: absolute;
			right: calc(50% - (38px / 2));
			top: calc(50% - (4px / 2));

			background-color: #349a89;

			border: none;
			border-radius: 10px;
			transition: all 0.1s;
		}

		&::after {
			transform: rotate(90deg);
		}
	}

	&[is-drag] {
		background-color: red;
	}

	& input {
		width: 100%;
		height: 100%;

		position: absolute;

		opacity: 0;

		cursor: inherit;
	}

	&-container {
		height: 100%;
		width: 100%;

		overflow: hidden;

		& > img {
			max-width: 100%;
		}
	}

	&-clear-button {
		height: 12px;
		width: 12px;
		position: absolute;

		right: -6px;
		top: -6px;

		background-color: #f77;

		border: 1px solid #f00;
		border-radius: 50%;

		transition: all 0.1s;

		&:hover {
			transform: scale(1.7);
			&::before,
			&::after {
				background-color: #00f;
			}
		}

		&::before,
		&::after {
			content: "";
			height: 80%;
			width: 1px;

			position: absolute;

			right: calc(50% - 7%);
			top: calc(50% - 38%);

			background-color: black;

			transform: rotate(45deg);
		}

		&::after {
			transform: rotate(-45deg);
		}
	}
}
