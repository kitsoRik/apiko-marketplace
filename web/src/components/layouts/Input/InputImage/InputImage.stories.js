import React from 'react';
import InputImage from './InputImage';

export const Default = () => <InputImage />

export default {
    title: "InputImage"
}