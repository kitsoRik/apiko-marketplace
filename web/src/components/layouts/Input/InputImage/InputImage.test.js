import React from 'react';
import { create } from 'react-test-renderer';
import InputImage from './InputImage';
import { mount } from 'enzyme';


describe("Label", () => {
    it("should be rendered", () => {
        const component = create(<InputImage />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it("should render with className", () => {
        const component = create(<InputImage className="my-custom-class-name" />);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it("should emit setFile", () => {
        const setFile = jest.fn();
        const testFile = new Blob(["image"], { type: 'image/png' });

        const component = mount(<InputImage setFile={setFile} />);

        component.find("input").simulate("change", { target: { files: [testFile] } });

        expect(setFile).toBeCalledWith(testFile);
    });
});