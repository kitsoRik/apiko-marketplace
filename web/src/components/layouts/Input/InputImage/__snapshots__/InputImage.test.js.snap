// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should be rendered 1`] = `
<div
  className="input-image "
  isDrag={null}
  onClick={[Function]}
>
  <input
    onChange={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    type="file"
  />
  <div
    className="input-image-container"
  >
    <img
      alt="Unknown"
      src={null}
    />
  </div>
</div>
`;

exports[`Label should render with className 1`] = `
<div
  className="input-image my-custom-class-name"
  isDrag={null}
  onClick={[Function]}
>
  <input
    onChange={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    type="file"
  />
  <div
    className="input-image-container"
  >
    <img
      alt="Unknown"
      src={null}
    />
  </div>
</div>
`;
